llvm/lib/CodeGen/MachineFunction.cpp

```diff
@@ -211,9 +211,8 @@ void MachineFunction::init() {
   ConstantPool = new (Allocator) MachineConstantPool(getDataLayout());
   Alignment = STI->getTargetLowering()->getMinFunctionAlignment();
 
-  // FIXME: Shouldn't use pref alignment if explicit alignment is set on F.
   // FIXME: Use Function::hasOptSize().
-  if (!F.hasFnAttribute(Attribute::OptimizeForSize))
+  if (!F.getAlign() && !F.hasFnAttribute(Attribute::OptimizeForSize))
```

Comment:
This will also disrespect getMinFunctionAlignment which is probably more of a problem

Reply:
Applying this in the function constructor also seems like the wrong place to do this

Reply:
I don't see why it doesn't respect `getMinFunctionAlignment`. `getMinFunctionAlignment` is read on line 212 and combined with the preferred alignment with `std::max` here, so the result will be at least `getMinFunctionAlignment`.